tensorflow_probability/python/mcmc/sample.py

```diff
@@ -308,8 +308,9 @@ def trace_everything(states, previous_kernel_results):
        http://statweb.stanford.edu/~owen/reports/bestthinning.pdf
   """
   is_seeded = seed is not None
+  if tf.executing_eagerly():
```

Comment:
Let's pull the `sample_chain` test into a separate PR.

A reasonable heuristic for sizing PRs is that a PR is the unit of roll-backs (at least internally).  If it was necessary to roll back the test of `sample_chain` for some reason, would it also be logical to roll back `step_kernel`?  To me the answer seems to be "no", so I recommend splitting them.

Also, of course, my comments about seeds apply here too.

Reply:
Sounds good!